### PR TITLE
Mark go packages which vendor directory can't be recreated broken

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

I don't really want to merge this as is but I hope I can get some attention by the maintainers to help fix the issues in these packages.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/tinygo/default.nix [hunk 1]
```diff
@@ -161,5 +161,8 @@ buildGoModule rec {
     description = "Go compiler for small places";
     license = licenses.bsd3;
     maintainers = with maintainers; [ Madouura muscaln ];
+    # vendor can't be recreated and fails with:
+    #   error: illegal path references in fixed-output derivation '/nix/store/......-tinygo-0.26.0-go-modules.drv'
```
Comment (kirillrdy) [inline pkgs/development/compilers/tinygo/default.nix:165]: @SuperSandro2000 can you explain how you got this error ? 

I recently tried updating tinygo and I kept getting vendor hash mismatch, so maybe its related

File: pkgs/development/compilers/tinygo/default.nix [hunk 2]
```diff
@@ -161,5 +161,8 @@ buildGoModule rec {
     description = "Go compiler for small places";
     license = licenses.bsd3;
     maintainers = with maintainers; [ Madouura muscaln ];
+    # vendor can't be recreated and fails with:
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/tinygo/default.nix:164]:
```suggestion
    # vendor can't be recreated by running `nix-build -A tinygo.go-modules --check` and fails with:
```

File: pkgs/development/compilers/tinygo/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/compilers/tinygo/default.nix:165]: just run the following ``nix-build -A tinygo.go-modules --check``

Comment (kirillrdy) [reply pkgs/development/compilers/tinygo/default.nix:165]: @SuperSandro2000 thanks for your hint, helped me to track down the problem https://github.com/NixOS/nixpkgs/pull/245445

**Comment by kirillrdy** [comment]
@SuperSandro2000 I've created PRs that should resolve reproducible vendor for those packages

**Comment by SuperSandro2000** [comment]
now we only need to remember to close this one, if they are all merged

**Pull request closed without merging**